### DISCUSSION TITLE
Idiomatic Rust vs C++ conformity

### DISCUSSION THREAD

**Discussion by peter-scholtens** [fltk-rs/fltk-rs#1067] [Ideas]
I wonder if it would not be more idiomatic Rust to safeguard this function:
```
fn Table::get_selection(&self) -> (i32, i32, i32, i32)
```
to something like:
```
fn Table::get_selection(&self) -> Option<(i32, i32, i32, i32)>
```
This would trigger the users to anticipate on what to do if nothing is selected (I made this error today).

**Comment by MoAlyousef** [comment]
Yeah. Working with tables is a bit strange at the moment, the reason being FLTK happily accepts negative numbers as valid selection, and will return them as well.
I've added TableExt::try_get_selection() and try_visible_cells() which return an Option instead. Changing the signature of set_selection() itself might break some user code.

**Reply by peter-scholtens** [reply]
Indeed, you want to keep the minor version updates and patches compatible. But for a major update this signature change could reduce the number of functions and make them more Rusty in general. That could be implemented with conditional compilation based on the version number of the library: in the same fltk 1.x use `fn x -> (...)` and `fn x_safe -> (...)` while in fltk 2.x use `fn x -> Option(...)` possibly with unwrap if the user prefers to do so. But do whatever you like, I only add this a suggestion.

**Reply by MoAlyousef** [reply]
I’ll replace the functions for version 2.0, I’ll add a reminder in the roadmap.
